File: py/IEC61968/Assets/Analytic.py
# Converted by an OPENAI API call using model: gpt-3.5-turbo-1106
from IEC61968.Assets.AnalyticKind import AnalyticKind
from IEC61968.Assets.AnalyticScore import AnalyticScore
from IEC61968.Assets.Asset import Asset
from IEC61968.Assets.AssetGroup import AssetGroup
from IEC61968.Assets.AssetHealthEvent import AssetHealthEvent
from IEC61968.Assets.ScaleKind import ScaleKind
from IEC61968.Common import Document


class Analytic(Document):
    
    def __init__(self):
        self.best_value = 0.0
        self.kind = AnalyticKind.OTHER
        self.scale_kind = ScaleKind.LINEAR
        self.worst_value = 0.0
        self.asset_group = AssetGroup()
        self.analytic_score = AnalyticScore()
        self.asset_health_event = AssetHealthEvent()
        self.asset = Asset()

File: java/IEC61968/Assets/Analytic.java
package IEC61968.Assets;

import IEC61970.Base.Domain.Float;
import IEC61968.Common.Document;

/**
 * An algorithm or calculation for making an assessment about an asset or asset
 * grouping for lifecycle decision making.
 * @author Gowri
 * @version 1.0
 * @created 25-Dec-2023 8:45:18 PM
 */
public class Analytic extends Document {

	/**
	 * Value that indicates best possible numeric value.
	 */
	public Float bestValue;
	/**
	 * Kind of analytic this analytic is.
	 */
	public AnalyticKind kind;
	/**
	 * The scoring scale kind.
	 */
	public ScaleKind scaleKind;
	/**
	 * Value that indicates worst possible numeric value.
	 */
	public Float worstValue;
	public AssetGroup AssetGroup;
	public AnalyticScore AnalyticScore;
	public AssetHealthEvent AssetHealthEvent;
	public Asset Asset;

	public Analytic(){

	}
}//end Analytic